### DISCUSSION TITLE
Mosaicing and cropping a set of images

### DISCUSSION THREAD

**Discussion by mrgeorge** [developmentseed/cogeo-mosaic#216] [Q&A] [answered]
I have a set of COGs that I'd like to merge with a simple pixel selection (first) and then crop with a geojson feature, returning a masked image at a native zoom level. This is doable with single COGs via the rio_tiler feature method, but is not implemented in the mosaic backend. Would extending the BaseBackend to implement the feature method be a reasonable way to pursue this? Is there a better way that I'm missing?

Thanks for your efforts on this repo and all the related tools!

**Comment by vincentsarago** [comment, marked as answer]
some previous discussion in titiler repo https://github.com/developmentseed/titiler/discussions/555

The `feature` method for mosaicjson could result in a lot of file request and thus give really poor result which is why we never added It in the Backend by default, in next version of cogeo-mosaic there will be new `asset_for_bbox` method in the backend which will ease the creation of such a function but I think we will still want this to be a custom function implemented by each user 🤔

**Reply by mrgeorge** [reply]
Ok thanks for the context @vincentsarago! My case has a small number of files overlapping, but with relatively frequent updates so merging into a single COG doesn't seem like a great option, and I'd like to have a pixel selection method for addressing the cases that do overlap, so I'll take a swing at implementing it for my own case.
